plugins/module_utils/foreman_helper.py

```diff
@@ -534,7 +540,11 @@ def resource_action(self, resource, action, params, options=None, data=None, fil
             msg = 'Error while performing {0} on {1}: {2}'.format(
                 action, resource, str(e))
             self.fail_from_exception(e, msg)
-        return True, result
+        if record_change:
+            if not ignore_check_mode:
```

Comment:
any reason not to fold these two `if`s into one?

Reply:
Well, it took me too long to figure, that the task waiting was calling `resource_action` and thereby recording a change (which was only a problem for the katello_manifest module). So i started to loose trust in my boolean logic. Then after that problem was fixed a few lines down, i thought: Hmm, this might be more readable.
But yes, i should probably merge it again.

Reply:
I do prefer one-line `if`s if possible ;)